CONTRIBUTING.md

```diff
@@ -1,7 +1,175 @@
+
+	1. On the 'spacetelescope/jwst' Github repository page, 'fork' the JWST repository to your own account space by clicking the appropriate button on the upper right-hand side. This will create an online clone of the main JWST repository but instead of being under the 'spacetelescope' organization, it will be under your personal account space. It is necessary to fork the repository so that many contributors aren't making branches directly on 'spacetelescope/jwst'. 
+
+	2. Now that you have an online ('remote') clone of `jwst`, it needs to be downloaded to your machine. To create this 'local' clone, choose an area on your file system and use the `git clone` command to clone your remote fork on to your machine: 
+		>> cd <directory>
+		>> git clone git@github.com:<your_username>/jwst.git`
+
+	3. Make sure that your references to 'origin' and 'upstream' are set correctly - you will need this to keep everything in sync and push your changes online. While your inital local clone will be an exact copy of your remote, which is an exact copy of the 'upstream' `spacetelescope/jwst`, these all must be kept in sync manually (via git fetch/pull/push). To check the current status of these references:
+		>> git remote -v
+
+	After your inital clone, you will likely be missing the reference to 'upstream' (which is just the most commonly used name in git to refer to the main project repository - you can call this whatever you want but the origin/upstream conventions are most commonly used) - to set this, use the `add` git command
+
+		If you are using an SSH key to authenticate.
+			>> git remote add upstream git@github.com:spacetelescope/jwst.git
+		Otherwise, you can simply set it to the repository URL but you will have to enter your password every time you fetch/push
+			>> git remote add upstream https://github.com/spacetelescope/jwst
+
+		If you ever want to reset these URLs, add references to other remote forks of `jwst` for collaboration, or change from URL to SSH, you can use the related `git remote set-url` command.
+
+### Step 2: Creating a branch for your changes
+
+It is a standard practice in git to create a new 'branch' (off 'master') for each new feature or bug fix. We will be creating a branch called 'my_feature' off of our local clone of the forked `jwst` repository:
+
+	1. Get your local 'master' branch up to date with any new changes upstream.
+		First, verify that you are currently on the `master` branch:
+			>> git status
+		and if your current branch isn't master,
+			>> git checkout master
+		Next, it is good practice to keep your local 'master' up-to-date with 'upstream/master'. To do this:
+			>> git pull upstream master
```

Comment:
I always thought that's a waste of effort, just forget about your local master, it's not useful for anything.

Reply:
good point, i dont know why im in this habit. removed.